### PR TITLE
Implement the Interface Adaptor Layer for Task

### PR DESCRIPTION
**Pull request by JoAndNight**
Description: This PR fixes the problems in implementation mentioned in PR#18. This PR also implemented the Interface Adaptor Layer and Framework & Drivers Layer for Task feature. The implementation is consistent with Clean Architecture.

- [ ]  Unit tests (For task feature): Before PR: 17% After PR: 17% Unit tests will be added in another PR linked to issue 27.
- [x]  Exception Handling: Exceptions are handled.
- [ ]  JavaDocs Documentation: More JavaDoc will be added in another PR linked to issue 27.
- [x]  SOLID principles
- [x]  Clean Architecture

### CODE REVIEW

**Review by MadhavKanna — APPROVED** [review]
approved!

File: src/main/java/controllers/tasks/CreateTaskController.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import use_cases.tasks.create_task.CreateTaskInputBoundary;
+import use_cases.tasks.create_task.CreateTaskInputData;
+import use_cases.tasks.create_task.CreateTaskOutputData;
+
+import java.util.Calendar;
+
+public class CreateTaskController {
+    final CreateTaskInputBoundary userInput;
+
+    public CreateTaskController(CreateTaskInputBoundary taskGateway) {
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/CreateTaskController.java:12]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/CreateTaskPresenter.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package controllers.tasks;
+
+import use_cases.tasks.create_task.CreateTaskOutputData;
+import use_cases.tasks.create_task.CreateTaskOutputBoundary;
+
+
+public class CreateTaskPresenter implements CreateTaskOutputBoundary {
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/CreateTaskPresenter.java:7]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/DeleteTaskController.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package controllers.tasks;
+
+import use_cases.tasks.delete_task.DeleteTaskInputBoundary;
+import use_cases.tasks.delete_task.DeleteTaskInputData;
+import use_cases.tasks.delete_task.DeleteTaskOutputData;
+
+public class DeleteTaskController {
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/DeleteTaskController.java:7]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/DeleteTaskPresenter.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package controllers.tasks;
+
+import use_cases.tasks.delete_task.DeleteTaskOutputBoundary;
+import use_cases.tasks.delete_task.DeleteTaskOutputData;
+
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/DeleteTaskPresenter.java:5]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/EditTaskController.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package controllers.tasks;
+
+import use_cases.tasks.edit_task.EditTaskInputBoundary;
+import use_cases.tasks.edit_task.EditTaskInputData;
+import use_cases.tasks.edit_task.EditTaskOutputData;
+
+import java.util.Calendar;
+
+public class EditTaskController {
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/EditTaskController.java:9]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/EditTaskController.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    final EditTaskInputBoundary userInput;
+
+    public EditTaskController(EditTaskInputBoundary taskGateway){this.userInput = taskGateway;}
+
+    public EditTaskOutputData edit(int taskId, String inputName, String category, Calendar deadline){
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/EditTaskController.java:14]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/EditTaskPresenter.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package controllers.tasks;
+
+import use_cases.tasks.edit_task.EditTaskOutputBoundary;
+import use_cases.tasks.edit_task.EditTaskOutputData;
+
+public class EditTaskPresenter implements EditTaskOutputBoundary{
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/EditTaskPresenter.java:6]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/MarkCompletionController.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package controllers.tasks;
+
+import use_cases.tasks.mark_task_completion.MarkCompletionInputBound;
+import use_cases.tasks.mark_task_completion.MarkCompletionInputData;
+import use_cases.tasks.mark_task_completion.MarkCompletionOutputData;
+
+public class MarkCompletionController {
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/MarkCompletionController.java:7]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/MarkVisibilityPresenter.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package controllers.tasks;
+
+import use_cases.tasks.mark_task_visibility.MarkVisibilityOutputBound;
+import use_cases.tasks.mark_task_visibility.MarkVisibilityOutputData;
+
+public class MarkVisibilityPresenter implements MarkVisibilityOutputBound{
+
+    @Override
+    public MarkVisibilityOutputData prepareSuccessView(MarkVisibilityOutputData outputData) {
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/MarkVisibilityPresenter.java:9]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/TaskCreationFailed.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package controllers.tasks;
+
+/**
+ * Exception deals with failure of creating task.
+ */
+public class TaskCreationFailed extends RuntimeException{
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/TaskCreationFailed.java:6]: Please add JavaDocs and documentation.

File: src/main/java/controllers/tasks/TaskEditionFailed.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package controllers.tasks;
+
+/**
+ * Exception deals the failure of task editing.
+ */
+public class TaskEditionFailed extends RuntimeException{
```
Comment (m-banthwan) [inline src/main/java/controllers/tasks/TaskEditionFailed.java:6]: Please add JavaDocs and documentation.

**Review by m-banthwan — APPROVED** [review]

File: src/main/java/use_cases/tasks/create_task/CreateTaskDsRequestModel.java [hunk 12]
```diff
@@ -1,29 +1,55 @@
...
+ */
 
 public class CreateTaskDsRequestModel {
-    private Task task;
-    private final LocalDateTime creationTime;
-
-    public CreateTaskDsRequestModel(Task task, LocalDateTime creationTime) {
-        this.task = task;
-        this.creationTime = creationTime;
+    private final String name;
```
Comment (EmilyDo) [inline src/main/java/use_cases/tasks/create_task/CreateTaskDsRequestModel.java:15]: if all these instance variables are final, shouldn't all of them be capitalized?

**Review by EmilyDo — APPROVED** [review]

**Pull request merged**